### PR TITLE
Update namespace transition instructions

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Proposed Changes

- Add namespace transition instructions for Workload Identity.
- No longer recommend copying secrets since it can lead to authentication errors.
- Removed the `yaml` and `shell` options since they are more likely to complicate the setup instead of simplifying it.

### CODE REVIEW

File: docs/how-to/namespace-transition.md
Comment (Harwayne) [inline docs/how-to/namespace-transition.md:181]: I think the BrokerCell only exists if there is at least one Broker using it.

Comment (Harwayne) [inline docs/how-to/namespace-transition.md:181]: Comment for line not edited in this PR:

```
1. Delete the old BrokerCell since it is no longer useful.
```

What happens if you just delete the BrokerCell itself? `kubectl delete brokercell default -n cloud-run-events` K8s GC may delete all the deployments, services, and HPAs for you.

Comment (Harwayne) [inline docs/how-to/namespace-transition.md:181]: Comment for line not edited in this PR:

When deleting the old source deployments, can we use a label selector, rather than making them do it manually? Do they all have the label `internal.events.cloud.google.com/controller=cloud-run-events-pubsub-pullsubscription-controller`?

Comment (Harwayne) [inline docs/how-to/namespace-transition.md:181]: Comment for line not edited in this PR:

```
# check what is left inside the old namespace
kubectl get all -n cloud-run-events
```

Are you sure that this will return `No resources found in cloud-run-events namespace.`? I have various things showing up (I believe they are all instances of non-namespaced CRDs).

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: fixed

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: I updated the instructions to copy the keys instead.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Removed the incorrect variable.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:174]: Updated to a single command.

Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:174]: Updated to a single command for all sources.

Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:174]: Removed the expected output comment.

Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:174]: Updated the docs to do this only if there are brokers deployed.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Added the variable.

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:46]: In GCP and K8s docs, individual shell commands are normally separated by additional steps that explain the desired state and give commands for verifying that state, so users are confident they know when to proceed and when to bail out of the process. In this case, since we are doing whole-cluster surgery, I think a bit more of that would help user confidence. https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:86]: Which `<gcp-project-identifier>` should I pick? If I have multiple projects, is this the project the cluster is in?

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:92]: These GSAs have default names, right? Most users will be using the defaults. Let's lead with those, and only proceed with looking at roles if the default names don't exist.

Also, we should give examples of the commands used to find the correct GSAs based on the list of roles. The user following these instructions may not be familiar with IAM commands. For example, I'm not familiar with IAM commands 😄

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:252]: Maybe replace "useful" with "being used" to clarify that this shouldn't cause a service disruption. Or be explicit that this won't cause a disruption.

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:291]:
```suggestion
   # delete all the source deployments referencing the old namespace
```

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:53]: The command uses a different delimiter than the standard syntax. It is the same functionality, but it avoids the need for escaping the `/`. I verified that it works correctly on MacOS as well.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Updated.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Done.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: I added a section for the default GSAs that were in the docs before the namespace change.

Unfortunately, I don't know how to do better than asking them to look for the correct GSAs by roles. The output of the `gcloud` command I already added is very verbose since it lists all the GSAs that exist in a project for each role. Using `grep` on the output is not particularly useful since the role is on a separate line at the bottom. In the future we will have a single role with a meaningful name and locating the GSAs will be a lot easier.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Added a comment before defining the variable.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Updated.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: Updated.

**Comment by AlexandraRoatis** [comment]
The document is ready for another review.

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:421]: Can we provide a command to list all sources in the cluster? Probably `kubectl get sources --all-namespaces` should work. We should also explain how to identify the ones that are GCP.

**Review by grantr — APPROVED** [review]
I love the new steps, thank you! Very clear. I have a couple nits but I'll lgtm anyway so this can get in. We can do any further changes in followups.

/lgtm

File: docs/how-to/namespace-transition.md
Comment (Harwayne) [inline docs/how-to/namespace-transition.md:456]:
```suggestion
     --selector internal.events.cloud.google.com/controller=cloud-run-events-pubsub-pullsubscription-controller
```

File: docs/how-to/namespace-transition.md
Comment (Harwayne) [inline docs/how-to/namespace-transition.md:127]: Make sure to also copy the Secret `google-cloud-sources-key`, if it is present.

That is the secret used by gcloud with `gcloud events init-namespace --copy-default-secret`.

**Comment by Harwayne** [comment]
/hold

At least until the sources deletion label selector is updated.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:45]: Got it! IntelliJ adds it automatically though (one click + zero typing).

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md:425]: Updated.

File: docs/how-to/namespace-transition.md
Comment (AlexandraRoatis) [reply docs/how-to/namespace-transition.md]: I made the correction.

**Review by Harwayne — COMMENTED** [review]
/lgtm

Leaving the hold in case @grantr wants to make any additional comments.

File: docs/how-to/namespace-transition.md
Comment (grantr) [inline docs/how-to/namespace-transition.md:435]: Cool, I didn't know we set this label

**Review by grantr — APPROVED** [review]
/lgtm

**Pull request merged**
